examples/autopilot_server/autopilot_server.cpp

```diff
@@ -0,0 +1,312 @@
+            std::cout << "No System Found from Autopilot" << std::endl;
+            return;
+        } else {
+            std::cout << "Setting System" << std::endl;
+        }
+        auto system = mavsdkTester.systems().back();
+        if (system == nullptr) {
+            std::cout << "System is null??" << std::endl;
+        }
+
+        // Create server plugins
+        auto paramServer = mavsdk::ParamServer{system};
+        auto telemServer = mavsdk::TelemetryServer{system};
+        auto actionServer = mavsdk::ActionServer{system};
+
+        // These are needed for MAVSDK at the moment
+        paramServer.provide_param_int("CAL_ACC0_ID", 1);
+        paramServer.provide_param_int("CAL_GYRO0_ID", 1);
+        paramServer.provide_param_int("CAL_MAG0_ID", 1);
+        paramServer.provide_param_int("SYS_HITL", 0);
+        paramServer.provide_param_int("MIS_TAKEOFF_ALT", 0);
+        // Add a custom param
+        paramServer.provide_param_int("my_param", 1);
+
+        // Allow the vehicle to change modes, takeoff and arm
+        actionServer.set_allowable_flight_modes({true, true, true});
+        actionServer.set_allow_takeoff(true);
+        actionServer.set_armable(true, true);
+
+        // Create a mission raw server
+        // This will allow us to receive missions from a GCS
+        auto missionraw = mavsdk::MissionRawServer{system};
+        std::cout << "MissionRawServer created" << std::endl;
+
+        auto mission_prom = std::promise<MissionRawServer::MissionPlan>{};
+        missionraw.subscribe_incoming_mission(
+            [&mission_prom](MissionRawServer::Result res, MissionRawServer::MissionPlan plan) {
+                std::cout << "Received Uploaded Mission!" << std::endl;
+                std::cout << plan << std::endl;
+                mission_prom.set_value(plan);
+            });
+        missionraw.subscribe_current_item_changed([](MissionRawServer::MissionItem item) {
+            std::cout << "Current Item Changed!" << std::endl;
+            std::cout << "Current Item: " << item << std::endl;
+        });
+        missionraw.subscribe_clear_all(
+            [](uint32_t clear_all) { std::cout << "Clear All Mission!" << std::endl; });
+        auto plan = mission_prom.get_future().get();
+        missionraw.set_current_item_complete();
+        missionraw.set_current_item_complete();
+        missionraw.set_current_item_complete();
+        missionraw.set_current_item_complete();
+
+        // Create vehicle telemetry info
+        TelemetryServer::Position position{55.953251, -3.188267, 0, 0};
+        TelemetryServer::PositionVelocityNed position_velocity_ned{{0, 0, 0}, {0, 0, 0}};
+        TelemetryServer::VelocityNed velocity{};
+        TelemetryServer::Heading heading{60};
+        TelemetryServer::RawGps rawGps{
+            0, 55.953251, -3.188267, 0, NAN, NAN, 0, NAN, 0, 0, 0, 0, 0, 0};
+        TelemetryServer::GpsInfo gpsInfo{11, TelemetryServer::FixType::Fix3D};
```

Comment:
Nitpick: if you want consistent variable naming :smile:

```suggestion
        TelemetryServer::RawGps raw_gps{
            0, 55.953251, -3.188267, 0, NAN, NAN, 0, NAN, 0, 0, 0, 0, 0, 0};
        TelemetryServer::GpsInfo gps_info{11, TelemetryServer::FixType::Fix3D};
```